spyder/plugins/ipythonconsole.py

```diff
@@ -871,6 +874,19 @@ def set_current_client_working_directory(self, directory):
             directory = encoding.to_unicode_from_fs(directory)
             shellwidget.set_cwd(directory)
 
+    def set_working_directory(self, dirname):
+        """Set current working directory.
+        In the workingdirectory and explorer plugins.
+        """
+        self.main.workingdirectory.chdir(dirname, refresh_explorer=True,
+                                         refresh_console=False)

```

Comment:
Tests are failing because we can't get cwd of external kernels, and the failing test is creating a console connected to an external kernel.

So please change this method to

```python
if dirname:
    self.main.workingdirectory.chdir(dirname, refresh_explorer=True,
                                     refresh_console=False)
```

I think with this change tests should pass.